EngineLayer/FdrAnalysis/PEPValueAnalysisGeneric.cs

```diff
@@ -78,40 +80,77 @@ public static string ComputePEPValuesForAllPSMsGeneric(List<PeptideSpectralMatch
             TrainTestData trainTestSplit = mlContext.Data.TrainTestSplit(dataView, testFraction: 0.1, null, 42);
             IDataView trainingData = trainTestSplit.TrainSet;
             IDataView testData = trainTestSplit.TestSet;
-
             var trainer = mlContext.BinaryClassification.Trainers.FastTree(labelColumnName: "Label", featureColumnName: "Features");
-
             var pipeline = mlContext.Transforms.Concatenate("Features", trainingVariables)
                 .Append(mlContext.BinaryClassification.Trainers.FastTree(labelColumnName: "Label", featureColumnName: "Features"));
+            var trainedModel = pipeline.Fit(trainingData); //this is NOT thread safe
+            var predictionEngine = mlContext.Model.CreatePredictionEngine<PsmData, TruePositivePrediction>(trainedModel);//this is NOT thread safe
```

Comment:
Should this be here? Aren't you training later after determining the threads? I don't see this engine used later.

Reply:
deleted line in question